lib/vehicle_factory.rb

```diff
@@ -0,0 +1,21 @@
+class VehicleFactory 
+
+    def initialize 
+    end
+
+    def create_vehicles(registration_locations)
+        registration_locations.each do |vehicle|
+            vehicle_details = ({vin: vehicle[:vin_1_10] , year: vehicle[:model_year], make: vehicle[:make], model: vehicle[:model], engine: [:ev]})
+            new_vehicles = Vehicle.new(vehicle_details)
+            return new_vehicles
+        end
+    end
+
+    def create_facility(facility_locations)
+        facility_locations.each do |facility|
+            facility_details = ({name: facility[:name], address: facility[:address], phone: facility[:phone]})
+            facility_locations = Facility.new(facility_locations)
+            return locations 
+        end 
+    end
+end
```

Comment:
I think you're off to a good start here. I'll attach a possible option for how this class could be set up: 

```ruby 
class VehicleFactory

  def create_vehicles(vehicle_dataset)
    vehicle_dataset.map do |vehicle_details|
      parsed_data = {
      vin: vehicle_details[:vin_1_10],
      year: vehicle_details[:model_year],
      make: vehicle_details[:make],
      model: vehicle_details[:model],
      engine: :ev
    }
      Vehicle.new(parsed_data)
    end
  end
end
```


Because there is no need to instantiate this object with arguments (i.e. we don't need to pass anything in when we call VehicleFactory.new), we don't need an initialize method. So, the one method that you see in this example is an instance method that can be called on any instance of the VehicleFactory class. 

```ruby
vehicle_factory = VehicleFactory.new
VehicleFactory.create_vehicles(data)
```